appender_integration_test.go

```diff
@@ -29,59 +29,110 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
-	"github.com/elastic/go-docappender/docappendertest"
-	"github.com/elastic/go-elasticsearch/v8"
-	"github.com/elastic/go-elasticsearch/v8/esapi"
-
 	"github.com/elastic/go-docappender"
+	"github.com/elastic/go-docappender/docappendertest"
+	elasticsearch7 "github.com/elastic/go-elasticsearch/v7"
+	esapi7 "github.com/elastic/go-elasticsearch/v7/esapi"
+	elasticsearch8 "github.com/elastic/go-elasticsearch/v8"
+	esapi8 "github.com/elastic/go-elasticsearch/v8/esapi"
 )
 
-func TestAppenderIntegration(t *testing.T) {
+const N = 100
+
+func sendEvents(t *testing.T, indexer *docappender.Appender, index string) {
+	for i := 0; i < N; i++ {
+		encoded, err := json.Marshal(map[string]any{"@timestamp": time.Now().Format(docappendertest.TimestampFormat)})
+		require.NoError(t, err)
+		err = indexer.Add(context.Background(), index, bytes.NewReader(encoded))
+		require.NoError(t, err)
+	}
+
+	// Closing the indexer flushes enqueued events.
+	err := indexer.Close(context.Background())
+	require.NoError(t, err)
+}
+
+func TestAppenderIntegrationV8(t *testing.T) {
```

Comment:
I don't think the testing is enough for the claim that we have tested with `v7`. I would suggest adding a wrapper for the tests that allow running ALL tests with both `v7` and `v8`. This would mean running tests for metric collection and other features that docappender has.

Reply:
Make sense. I can remove the comment about v7 support.